dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/FunctionChoice.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represents an AI model's decision-making strategy for calling functions, offering predefined choices: Auto, Required, and None.
+/// Auto allows the model to decide if and which functions to call, Required enforces calling one or more functions, and None prevents any function calls, generating only a user-facing message.
+/// </summary>
+public readonly struct FunctionChoice : IEquatable<FunctionChoice>
+{
+    /// <summary>
+    /// This choice instructs the model to decide whether to call the functions or not and, if so, which ones to call.
+    /// </summary>
+    public static FunctionChoice Auto { get; } = new("auto");
```

Comment:
Is the expectation that if an AI connector implementation supports Auto, but calls it something like "automatic", we will still be able to use it with a behavior that returns a FunctionChoice of Auto?

Is the idea that common behaviors will be listed here, but uncommon ones will be available from each connector implementation?

Reply:
I think in case of different names, on connector level there will be some kind of mapping between connector choice name and SK choice name.

Reply:
Yep, each connector is going to a corresponding mapping.